lib/connections/connection.js

```diff
@@ -0,0 +1,146 @@
+'use babel';
+
+import Logger from '../helpers/logger';
+import ApiWrapper from '../wrappers/api-wrapper';
+
+export default class Connection {
+  constructor(device, pyboard, settings) {
+    this.pyboard = pyboard;
+    this.terminal = device.terminal;
+    this.settings = settings;
+    this.device = device;
+    this.view = this.device.view;
+    this.api = new ApiWrapper(settings);
+    this.logger = new Logger('ConnectionHelper');
+    this.message_callback = null;
+    this.connection_timer = null;
+  }
+
+  setMessageCallback(cb) {
+    this.message_callback = cb;
+  }
+
+  connectedInOtherWindow(address) {
+    const state = this.api.getConnectionState(address);
+    const ts = new Date().getTime();
+    if (
+      state &&
+      state.project !== this.view.project_name &&
+      state.timestamp > ts - 11000
+    ) {
+      return state.project;
+    }
+    return null;
+  }
+
+  connect(cb, onDisconnect) {
+    const _this = this;
```

Comment:
I think you not might need to do this. Please try to change this. 
`this` should still point the class object (as long as you're using arrow functions)